File: Leetcode/cpp/medium/1980.cpp
/*
 * @lc app=leetcode.cn id=1980 lang=cpp
 *
 * [1980] 找出不同的二进制字符串
 */


// @lcpr-template-start
#include <bits/stdc++.h>
using namespace std;
// @lcpr-template-end
// @lc code=start
#include <ranges>

class Solution {
public:
    string findDifferentBinaryString(vector<string>& nums) {
        int n = nums.size();
        string ans(n, '0');
        for (auto const [i, num] : views::enumerate(nums))
            ans[i] = num[i] == '0' ? '1' : '0';
        return ans;
    }
};
// @lc code=end

File: Leetcode/rust/medium/1980.rs
/*
 * @lc app=leetcode id=1980 lang=rust
 *
 * [1980] Find Unique Binary String
 */

// @lcpr-template-start
struct Solution;
// @lcpr-template-end
// @lc code=start
impl Solution {
    pub fn find_different_binary_string(nums: Vec<String>) -> String {
        nums.iter()
            .enumerate()
            .map(|(i, num)| if num.as_bytes()[i] == b'0' { '1' } else { '0' })
            .collect()
    }
}
// @lc code=end
